### PR TITLE
WFS 2.0.0 support

### PR DESCRIPTION
**Pull request by bjornharrtell**
ref https://github.com/openlayers/openlayers/issues/11410 and https://github.com/openlayers/openlayers/issues/11411

Remaining work:

* [x] Default to GML 3.2 if WFS 2.0.0 is selected
* [x] Verify/test GetFeature request write
* [x] Verify/test GetFeature response read
* [x] Verify/test Transaction write
* [x] Adjust documentation

### CODE REVIEW

**Comment by bjornharrtell** [comment]
@ahocevar this is now ready for review.

**Comment by tschaub** [comment]
Thanks for your work on this, @bjornharrtell.

File: src/ol/format/WFS.js [hunk 1]
```diff
@@ -262,7 +321,7 @@ class WFS extends XMLFeature {
   readFeaturesFromNode(node, opt_options) {
     /** @type {import("../xml.js").NodeStackItem} */
     const context = {
-      node: node,
+      node,
```
Comment (tschaub) [inline src/ol/format/WFS.js:324]: I didn’t confirm, but I’m assuming the shorthand gets transpiled away for IE.

Comment (bjornharrtell) [reply src/ol/format/WFS.js:324]: Also my conclusion but also not confirmed.

**Comment by bjornharrtell** [comment]
Thanks for review and merge @tschaub.

**Pull request merged**